### PR TITLE
add go lint rule in Makefile

### PR DESCRIPTION
**Pull request by VineethReddy02**
We have the go-lint checks in place in CI. But we do not have it in Makefile to test the linting locally before committing. So added it into Makefile for a better development experience.

### CODE REVIEW

File: config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml [hunk 1]
```diff
@@ -282,6 +282,7 @@ spec:
                 type: string
               tolerations:
                 description: Toleration to schedule OpenTelemetry Collector pods.
+                  This is only relevant to daemonsets, statefulsets and deployments
```
Comment (jpkrohling) [inline config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml:285]: Could you revert this change? Otherwise it might conflict with another PR that is fixing this.

Comment (VineethReddy02) [reply config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml:285]: I didn't see this in other PRs. I mean even in your recent PR (you reverted it, so I was thinking to add it with a separate PR).

Comment (jpkrohling) [reply config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml:285]: This is being changed here: https://github.com/open-telemetry/opentelemetry-operator/pull/306/files#diff-c5e8337ee47f0f73be5c1b6eb6e52fff3c9cde0a18b59d696dc827e1fba75e10R285

**Review by jpkrohling — APPROVED** [review]
